chromadb/utils/async_to_sync.py

```diff
@@ -0,0 +1,62 @@
+import inspect
+import asyncio
+from typing import Any, Callable, Coroutine, ParamSpec, TypeVar
+
+
+P = ParamSpec("P")
+R = TypeVar("R")
+
+
+def async_to_sync(func: Callable[P, Coroutine[Any, Any, R]]) -> Callable[P, R]:
+    """A function decorator that converts an async function to a sync function.
+
+    This should generally not be used in production code paths.
+    """
+
+    def sync_wrapper(*args, **kwargs):  # type: ignore
+        loop = None
+        try:
+            loop = asyncio.get_event_loop()
+        except RuntimeError:
+            loop = asyncio.new_event_loop()
+            asyncio.set_event_loop(loop)
+
+        if loop.is_running():
+            return func(*args, **kwargs)
+
+        result = loop.run_until_complete(func(*args, **kwargs))
+
+        def convert_result(result: Any) -> Any:
```

Comment:
pretty hacky, but need a way to automatically transform returned `AsyncCollection()` instance methods to sync methods